x-pack/plugins/apm/public/utils/formatters.js

```diff
@@ -52,3 +52,14 @@ export function asInteger(value) {
 export function tpmUnit(type) {
   return type === 'request' ? 'rpm' : 'tpm';
 }
+
+export function getDurationPercent(
+  itemDuration: number,
+  totalDuration?: number,
+  fallbackResult?: string
+) {
+  if (totalDuration === undefined || totalDuration === 0) {
+    return fallbackResult || '';
+  }
+  return numeral(itemDuration / totalDuration).format('0.00%');
```

Comment:
Curious what Soren thinks, but I *think* I'd prefer a "safeDivide" function and let the caller decide if/how to format number of decimal places, formatting, etc. since those may be different in different contexts? Maybe that's premature but trying to find the balance between doing things too early and sensible defaults etc...

Reply:
Perhaps we should have both `getDurationPercent` and `safeDivide` the the former calls the later. I see `getDurationPercent` used several places, so it probably makes sense to have the formatting logic in one place.

Reply:
On the other hand, in the above example, it might be better to just add the following at the top `WaterfallItem`, since it doesn't make sense to render the component without `totalDuration`:

```diff
+   if(!totalDuration) {
+     return null;
+   }
    const width = (item.duration / totalDuration) * 100;
    const left = (item.offset / totalDuration) * 100;
```